### ISSUE TITLE
phpstorm not detecting tests in parallel run

### ISSUE THREAD

**Issue by ariaieboy** [pestphp/pest#552]
when i run the tests using `-p` option, the phpstorm not detecting the tests and return "Tests passed: 0" after running tests in parallel mode. while i have 160 passed test in RUN log and if i remove the -p option it will detect the tests currectly.

**Comment by ariaieboy** [comment]
@olivernybroe sorry for the late response I didn't get a notification about your reply.

![image](https://user-images.githubusercontent.com/15873972/184194840-d44387ba-45e6-472d-8542-f985fabffc3c.png)

```
brianium/paratest                            v6.6.1
pestphp/pest-plugin-parallel                 v1.2.0 
pestphp/pest                                 v1.21.3 
phpunit/phpunit                              9.5.21 
```

**Comment by ariaieboy** [comment]
> @olivernybroe sorry for the late response I didn't get a notification about your reply.

this the command that phpstorm runs 
`php /var/www/html/vendor/pestphp/pest/bin/pest --teamcity -p --configuration /var/www/html/phpunit.xml`

it shows tests styled :
![image](https://user-images.githubusercontent.com/15873972/184195870-37e237e4-b9b7-4b11-a642-e1932902d6fd.png)

in this picture it's obvious that more than 200 tests has been passed but phpstorm didnt detect any of it:
![image](https://user-images.githubusercontent.com/15873972/184196044-025de103-b663-420b-a15d-1ea9d525b6f3.png)

**Comment by olivernybroe** [comment]
Huh, weird that it is able to run for you at all with those arguments, but yeah the issue is that the `teamcity` printer is not taken affect when having parallel flag on and without that output format, I cannot detect it in phpstorm.

Therefore the solution here is that pest needs to support teamcity printer in parallel also 👍

**Comment by ariaieboy** [comment]
in this line [PestRunnerWorker.php#L84](https://github.com/pestphp/pest-plugin-parallel/blob/264f8086f2d0b4a7e657cf5ca48de6ed5587332a/src/Paratest/PestRunnerWorker.php#L84) parallel plugin will force the default printer. 

I changed this line with this code :
```PHP
        if ($options->teamcity()){
            $args[$printerIndex] = TeamCity::class;
        }else{
            $args[$printerIndex] = Printer::class;
        }
```
and I managed to get TeamCity output from the parallel plugin. 
but it has 2 problems:
first, it prints a lot of spam messages like this :
![image](https://user-images.githubusercontent.com/15873972/184445796-65f0b30b-6246-4b78-861c-d9f0d66b3139.png)
and the second problem is that it shows the test's like this:
![image](https://user-images.githubusercontent.com/15873972/184446020-5fc29c69-139d-4e13-a644-1e8bba12a467.png)

I think it's a good starting point to add TeamCity support to the parallel plugin.

I would love to help with this issue, but I need some tips on how can i fix this problem. because I don't have enough knowledge about TeamCity formatting and pest-parallel internals.

cc: @nunomaduro
